src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -580,6 +610,62 @@ def post_process_containerapp(self, containerapp_def, r):
                 linker_client.linker.begin_create_or_update(resource_uri=r["id"],
                                                             parameters=item["parameters"],
                                                             linker_name=item["linker_name"]).result()
+
+        if self.get_argument_repo():
+            app = self.set_up_create_containerapp_if_source_or_repo(containerapp_def=containerapp_def)
+            r = self.set_up_create_containerapp_repo(app=app, r=r, env=app.env, env_rg=app.resource_group.name)
+        return r
+
+    def set_up_create_containerapp_if_source_or_repo(self, containerapp_def):
```

Comment:
Hi, I have refactor the containerapp decorator with this pr: [containerapp refactor decorator by Greedygre · Pull Request #6511 · Azure/azure-cli-extensions (github.com)](https://github.com/Azure/azure-cli-extensions/pull/6511)
You can use `self.containerapp_def` to set and get current payload for containerapp.
Thanks for using this.